crates/red_knot/src/symbols.rs

```diff
@@ -457,19 +485,31 @@ impl PreorderVisitor<'_> for SymbolTableBuilder {
                 level,
                 ..
             }) => {
+                let module = module.as_ref().map(|m| Name::new(&m.id));
+
                 for alias in names {
                     let symbol_name = if let Some(asname) = &alias.asname {
                         asname.id.as_str()
                     } else {
                         alias.name.id.as_str()
                     };
                     let def = Definition::ImportFrom(ImportFromDefinition {
-                        module: module.as_ref().map(|m| m.id.clone()),
-                        name: alias.name.id.clone(),
+                        module: module.clone(),
+                        name: Name::new(&alias.name.id),
                         level: *level,
                     });
                     self.add_symbol_with_def(symbol_name, def);
                 }
+
+                if let Some(level) = *level {
```

Comment:
Our AST is confusing/redundant, because it has level as an `Option`, but it is never `None`. An import like `from foo import bar` (no dots) is represented in our AST with `level` as `Some(0)`, not `None`.

So if we leave the AST as-is, this code needs to be fixed to also push a `Module` dependency if `level` is `Some(0)`. I think we should fix the AST, though -- it should either not use `Option` at all, or it should set `None` instead of `Some(0)`. (I think the latter is clearer; removing the `Option` would save 4 bytes, but I don't think that matters since `StmtImportFrom` is not close to being the largest statement.)

Reply:
Thanks for resolving this for me :)